dvc/command/experiments.py

```diff
@@ -299,12 +317,18 @@ def experiments_table(
     sort_by=None,
     sort_order=None,
     precision=DEFAULT_PRECISION,
+    include_state=True,
+    include_executor=True,
 ) -> "TabularData":
     from funcy import lconcat
 
     from dvc.compare import TabularData
 
-    headers = ["Experiment", "rev", "queued", "typ", "Created", "parent"]
+    headers = ["Experiment", "rev", "typ", "Created", "parent"]
+    if include_state:
+        headers.append("State")
+    if include_executor:
+        headers.append("Executor")
```

Comment:
I wanted to get rid of these kinds of conditional checks in the table. And, it is not clear on what basis `include_state` and `include_executor` are set/works.

I'll suggest to get rid of these conditionals, and add the following in the header itself. 
```python
headers = ["Experiment", "rev", "typ", "Created", "parent", "State", "Executor"]
```

Also, get rid of all the `if` checks and keep everything in the table. Later you can simply drop them if they are empty (after [here](https://github.com/iterative/dvc/blob/61835f65116c3e5808c325d6cb22a12d36f70cde/dvc/command/experiments.py#L389)).

```python
if td.is_empty("State"):
    td.drop("State")
if td.is_empty("Executor"):
    td.drop("Executor")
```

And, in the implementation of `TabularData`, you could have something like this:
```python
def is_empty(self, name: str) -> bool:
    col = self.column(name)
    return not any(item != self._fill_value for item in col)
```

This should clean up the code here, and also make it safe and clear.